packages/core/admin/ee/server/register.js

```diff
@@ -12,6 +13,33 @@ module.exports = async ({ strapi }) => {
     strapi.container.register('audit-logs', auditLogsService);
     await auditLogsService.register();
   }
-
+  if (features.isEnabled('review-workflows')) {
+    addReviewWorkflowMiddleware(strapi);
+  }
   await executeCERegister({ strapi });
 };
+
+/**
+ * A Strapi middleware function that adds support for review workflows.
```

Comment:
Nice idea to use a middleware for this :)

And a minor comment. As a dev looking at this, I would lack info about why we do this.
Wdyt about explaining the frontend can only send the RW at the top level, so we have to move it here?